src/libsystemd/sd-varlink/sd-varlink.c

```diff
@@ -1576,9 +1570,8 @@ _public_ int sd_varlink_get_current_parameters(sd_varlink *v, sd_json_variant **
                 return -ENODATA;
 
         p = sd_json_variant_by_key(v->current, "parameters");
-        if (!p)
-                return -ENODATA;
 
+        /* For backward compatibility, return NULL when parameters field is missing (empty case) */
```

Comment:
i don't understand the comment here? we should always return an emty object here as before